src/WebViewShared.tsx

```diff
@@ -44,11 +44,9 @@ const createOnShouldStartLoadWithRequest = (
     const { url, lockIdentifier } = nativeEvent;
 
     if (!passesWhitelist(compileWhitelist(originWhitelist), url)) {
-      Linking.openURL(url);
+      Linking.openURL(url).catch(() => {});
```

Comment:
Would this be cleaner with `async/await`?

Not a fan of silent errors, should at least warn in dev mode what happened.

Reply:
Re: `async/await` I don't _think_ so because the function this is in is not `async` currently (or returning a promise), and we probably(?) don't want to block the rest of the function (calling `loadRequest`) on `await Linking.openURL`... that said I am not 100% sure and am not familiar with the code in any detail so I could totally be wrong.

Re: silent errors, yeah that makes sense.  I uh, borrowed this "pattern" from a markdown library that handles `Linking.openURL` errors this way (eating it) :) I see a handful of places in this library where we call `console.warn`.  Would just doing that be sufficient (e.g. `.catch(e => console.warn('Error calling Linking.openURL', e))`)?

Reply:
Sorry about that, for some reason I thought I read `reject` 😂 ignore `async/await`.

So there is a way to pre-check if the app can open a url: https://facebook.github.io/react-native/docs/linking#opening-external-links, this might be useful.

`console.warn` looks good to me, that triggers the YellowBox.

Reply:
Good find, I updated the code to use `canOpenURL` before calling `openURL.   This required some changes to the tests because they need to wait for the Promise returned by `canOpenURL` to resolve.  I added a pretty good sized comment to the tests hopefully explaining that.  Hopefully that is not too gross.

I also added the code to call `console.warn` on errors instead of eating them.  I wrapped it in a `if (__DEV__)` check so they don't get run on release builds, though its probably not a big deal.  I had to updated `.eslintrc` to make `ESLint` ok with using `__DEV__`.  I think that is standard practice but I am happy to revert if it seems wrong.

Reply:
Thanks for all this work!

I would say remove `__DEV__`, that should be developers who use this package responsibility to configure their metro configs to handle.

With that change, LGTM. Will ping the team to check it out

Reply:
OK updated.  Thanks for the feedback.

BTW for my education can you tell me what you mean by `configure their metro configs to handle.`?  Is there an option to auto remove calls to the `console.*` functions?

Reply:
Yup! Thanks for the change.

Check this out:
https://facebook.github.io/react-native/docs/performance.html#using-consolelog-statements